### PR TITLE
[Docs] New website

### PR DESCRIPTION
**Pull request by jpdriver**
### Description

- As part of Formidable's ongoing commitment to Open Source, we are starting to standardize our OSS Landers around [docusaurus](https://docusaurus.io).
- This PR replaces the existing `react-static` site with a new one backed by `docusaurus`.
- Markdown content is kept separately at `/docs` and all docusaurus-related code is located in the new `/website` folder.
- This is largely based on prior work done for [react-native-owl](https://github.com/FormidableLabs/react-native-owl/tree/main/website)

#### Type of Change

- [x] This change requires a documentation update

### How Has This Been Tested?

Local testing through `yarn start` in `/website`.

### Checklist: (Feel free to delete this section upon completion)

- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by ryan-roemer** [comment]
Checked out https://formidable-com-spectacle-staging-1109.surge.sh/open-source/spectacle/ a bit and looks great so far!!!

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -1,82 +1,64 @@
-name: Docs
+name: Deploy Website
 
-defaults:
-  run:
-    working-directory: ./docs
```
Comment (ryan-roemer) [inline .github/workflows/docs.yml:5]: Do we want to restore this for `website` and remove all the individual `working-directory: ./website` configs for each script command below?

File: website/docusaurus.config.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  presets: [
+    [
+      'classic',
+      /** @type {import('@docusaurus/preset-classic').Options} */
+      ({
+        docs: {
+          path: '../docs',
+          sidebarPath: require.resolve('./sidebars.js'),
+          editUrl:
+            'https://github.com/FormidableLabs/spectacle/tree/main/website',
```
Comment (ryan-roemer) [inline website/docusaurus.config.js:29]: Is `editUrl` `docs` or `website`?

File: website/src/components/home/preview.tsx [hunk 3]
```diff
@@ -31,7 +25,7 @@ export default function Preview({ preview }): JSX.Element {
         target="_blank"
         rel="noopener noreferrer"
       >
-        <Video muted loop poster={preview.bgStill}>
+        <Video autoPlay muted loop poster={preview.bgStill}>
```
Comment (jpdriver) [inline website/src/components/home/preview.tsx:28]: could be worth adding an `engines` limit to `package.json` to prevent anyone else hitting this?

Comment (ryansrofe) [reply website/src/components/home/preview.tsx]: @jpdriver this has been added in my last batch of code as we discussed

File: website/docusaurus.config.js (hunk 2 above)
Comment (gksander) [reply website/docusaurus.config.js:66]: We have this in docusaurus config:

```
editUrl: 'https://github.com/FormidableLabs/spectacle/tree/main/website',
path: '../docs',
```

With the addition of `'../docs'`, I have confirmed that these edit links do indeed point to the right place (which is the `docs` folder).

File: website/src/components/index/hero.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import styles from './hero.module.scss';
+
+type NavItem = {
+  text: string;
+  url: string;
+};
+
+function NavItem({ text, url }: NavItem) {
+  const isExternalUrl = /^(http|https):\/\//.test(url);
+  const isDocsUrl = /^docs\//.test(url);
```
Comment (gksander) [inline website/src/components/index/hero.tsx:15]: Might consider using [URL API](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) to do these checks. (May or may not be worth the change.)

File: website/src/components/index/hero.module.scss [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+
+        @media ($media-breakpoint-md) {
+          margin-top: 0;
+          width: $sizing-xxs;
+        }
+
+        &:hover {
+          color: var(--oss-text-light);
+          background-position: 100%;
+          // background: var(--oss-button-light-hover);
```
Comment (gksander) [inline website/src/components/index/hero.module.scss:139]:
```suggestion
```

File: website/src/components/index/get-started.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+import Wrapper from '@site/src/components/global/wrapper';
+import styles from './get-started.module.scss';
+
+interface Meta {
+  theme: string;
+  noMargin: boolean;
+  noPadding: boolean;
+}
+
+export default function GetStarted({ content }): JSX.Element {
```
Comment (gksander) [inline website/src/components/index/get-started.tsx:15]:
```suggestion
export default function GetStarted({ content }) {
```

File: website/src/components/global/body-copy.tsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  theme: string;
+  children: React.ReactNode;
+};
+
+const BodyCopy = ({ className, theme, children }: BodyCopy) => {
+  const isLight = theme === 'Light' || theme === 'Color';
+  const classNames = clsx(
+    styles.bodyCopy,
+    isLight && styles.light,
+    className && className
```
Comment (gksander) [inline website/src/components/global/body-copy.tsx:16]:
```suggestion
    className
```

File: website/src/components/global/button.module.scss [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+  min-width: $sizing-sm;
+  text-align: center;
+  text-transform: uppercase;
+  width: auto;
+  padding-left: $spacing-xs;
+  padding-right: $spacing-xs;
+  font-weight: $font-weights-link;
+
+  &:hover {
+    // background: var(--oss-button-light-hover);
```
Comment (gksander) [inline website/src/components/global/button.module.scss:33]:
```suggestion
```

File: website/src/components/global/button.module.scss [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+        var(--oss-button-hover) 50%,
+        transparent 50%
+      );
+      position: 0;
+      size: 400%;
+    }
+    color: var(--oss-text-light);
+
+    &:hover {
+      // background: var(--oss-button-hover);
```
Comment (gksander) [inline website/src/components/global/button.module.scss:60]:
```suggestion
```

File: website/src/components/global/npm-copy.module.scss [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+  color: var(--oss-text-dark);
+  border: 0;
+  border-radius: 0 var(--oss-border-radius) var(--oss-border-radius) 0;
+  text-transform: uppercase;
+  cursor: copy;
+
+  &:hover {
+    color: var(--oss-text-light);
+    background-position: 100%;
+    // background: var(--oss-button-light-hover);
```
Comment (gksander) [inline website/src/components/global/npm-copy.module.scss:54]:
```suggestion
```

File: website/src/components/global/secondary-title.tsx [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+  theme: string;
+  children: React.ReactNode;
+};
+
+const SecondaryTitle = ({ className, theme, children }: SecondaryTitle) => {
+  const isLight = theme === 'Light' || theme === 'Color';
+  const classNames = clsx(
+    styles.secondaryTitle,
+    isLight && styles.light,
+    className && className
```
Comment (gksander) [inline website/src/components/global/secondary-title.tsx:16]:
```suggestion
    className
```

File: website/src/pages/index.module.scss [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+
+body {
+  font-family: '-apple-system', 'BlinkMacSystemFont', 'Segoe UI', 'Roboto',
+    'Helvetica Neue', 'Arial', 'Noto Sans', 'sans-serif', 'Apple Color Emoji',
+    'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
+  text-rendering: optimizeLegibility;
+
+  @media ($media-breakpoint-lg) {
+    // font-size: $font-sizes-body;
+  }
```
Comment (gksander) [inline website/src/pages/index.module.scss:21]:
```suggestion
```

Doesn't seem like we're using this.

**Review by gksander — COMMENTED** [review]
This is a chonky PR, but this thing is _slick_. I dropped a few comments here for us to address real quick, but nothing major. I think we do a little bit of tidying up (I can actually get started on that), get a final design stamp of approval on this, and ship this thing!

File: .github/workflows/docs.yml (hunk 1 above)
Comment (gksander) [reply .github/workflows/docs.yml:5]: I like this idea.

**Review by gksander — APPROVED** [review]
LGTM, LFG 🚀

**Pull request merged**